java/semantickernel-core-skills/src/main/java/com/microsoft/semantickernel/coreskills/HttpSkill.java

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+package com.microsoft.semantickernel.coreskills;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.netty.NettyAsyncHttpClientBuilder;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * A skill that provides HTTP functionality.
+ *
+ * <p>Usage:
+ *
+ * <p>Usage: kernel.importSkill(new HttpSkill(), "http");
+ *
+ * <p>Examples:
+ *
+ * <p>{{http.getAsync $url}}
+ *
+ * <p>{{http.postAsync $url}}
+ *
+ * <p>{{http.putAsync $url}}
+ *
+ * <p>{{http.deleteAsync $url}}
+ */
+public class HttpSkill {
+
+    private final HttpClient httpClient;
+
+    public HttpSkill() {
+        this.httpClient = new NettyAsyncHttpClientBuilder().build();
```

Comment:
It is perhaps better to reference HttpClient.createDefault() so that it uses the SPI mechanism of the Azure SDK to load an Http Client.

This way, we don't need to have a dependency on the Netty one, and users can load whatever they want.

https://github.com/Azure/azure-sdk-for-java/wiki/HTTP-clients/0b5cb17511eb25be4ab7b6a78c765d27748fa907#configuring-http-clients

Reply:
Does the project still builds and the unit test for this skill passes without this dependency on test scope?

Reply:
Excellent!